### PR TITLE
ING-747: Support no expiry for collections

### PR DESCRIPTION
**Pull request by Westwooo**
WIP - needs associated changes in gocbcorex

### CODE REVIEW

**Comment by brett19** [comment]
Code looks good, but there is a merge commit that shouldn't exist.

**Comment by Westwooo** [comment]
> Code looks good, but there is a merge commit that shouldn't exist.

Squashed the commits.

File: gateway/apiversion/apiversion.go [hunk 1]
```diff
@@ -2,8 +2,9 @@ package apiversion
 
 type ApiVersion uint64
 
-const Latest ApiVersion = 20240510
+const Latest ApiVersion = QueryDurabilityLevel
```
Comment (chvck) [inline gateway/apiversion/apiversion.go:5]: Should this be bumped?

File: gateway/dataimpl/server_v1/collectionadminserver.go [hunk 2]
```diff
@@ -147,14 +163,28 @@ func (s *CollectionAdminServer) CreateCollection(
 	ctx context.Context,
 	in *admin_collection_v1.CreateCollectionRequest,
 ) (*admin_collection_v1.CreateCollectionResponse, error) {
+	apiVersion, err := apiversion.FromContext(ctx)
```
Comment (chvck) [inline gateway/dataimpl/server_v1/collectionadminserver.go:166]: Might be nice to have a helper function which takes a version and returns if the version is supported by the client. Not needed to for this PR though.

File: gateway/apiversion/apiversion.go (hunk 1 above)
Comment (Westwooo) [reply gateway/apiversion/apiversion.go]: It should, good catch. Done now

**Review by chvck — APPROVED** [review]

**Pull request merged**